### ISSUE TITLE
unable to use find_in_parent_folders to find folders

### ISSUE THREAD

**Issue by Macaber** [gruntwork-io/terragrunt#2868]
**Describe the bug**

```
terragrunt apply                                                                                                                                                                              14:38:17
ERRO[0000] /tfwork/tg/terragrunt.hcl:2,15-38: Error in function call; Call to function "find_in_parent_folders" failed: ParentFileNotFound: Could not find a tfwork in any of the parent folders of /tfwork/tg/terragrunt.hcl. Cause: Traversed all the way to the root.., and 1 other diagnostic(s) 
ERRO[0000] Unable to determine underlying exit code, so Terragrunt will exit with error code 1 
```

path tree
```
tfwork
├── test-var
│   ├── aaa.sh.tftpl
│   ├── aaa.tftpl
│   ├── main.tftest.hcl
│   ├── providers.tf
│   ├── terraform.tfstate
│   ├── terraform.tfstate.backup
│   └── variables.tf
└── tg
    └── terragrunt.hcl
```

**To Reproduce**
Steps to reproduce the behavior, code snippets and examples which can be used to reproduce the issue.

```hcl
include "module" {
    path = "${find_in_parent_folders("tfwork")}/test-var/providers.tf"
}

```

**Expected behavior**
can read providers.tf file

**Nice to have**
- [x] Terminal output
- [ ] Screenshots

**Versions**
- Terragrunt version:0.54.11
- Terraform version:v1.5.7
- Environment details (Ubuntu 20.04, Windows 10, etc.): mac m1

**Additional context**

**Comment by Macaber** [comment]
aaa.tf
```
variable "orchestrator_path" {
  type = string
}

output "name" {
  value = var.orchestrator_path
}
```
terragrunt.hcl
```
inputs = {
  orchestrator_path        = find_in_parent_folders("tfwork")
}
```
tree path
```
tfwork
├── test-var
│   ├── aaa.sh.tftpl
│   ├── aaa.tftpl
│   ├── main.tftest.hcl
│   ├── providers.tf
│   ├── terraform.tfstate
│   ├── terraform.tfstate.backup
│   └── variables.tf
└── tg
    ├── aaa.tf
    └── terragrunt.hcl
```

terminal output
```
~/mywork/tfwork/tg ❯ terragrunt apply                                                                                                                                                                                                                                                
ERRO[0000] /Users/yfsun/mywork/tfwork/tg/terragrunt.hcl:6,30-53: Error in function call; Call to function "find_in_parent_folders" failed: ParentFileNotFound: Could not find a tfwork in any of the parent folders of /Users/yfsun/mywork/tfwork/tg/terragrunt.hcl. Cause: Traversed all the way to the root.. 
```

**Comment by Macaber** [comment]
i can read path in past version such as 0.48.5, but now i upgrade terragrunt to 0.54.11, find_in_parent_folders can't work

**Comment by oscarmendoza123** [comment]
+1

**Comment by levkohimins** [comment]
Resolved in [v0.54.12](https://github.com/gruntwork-io/terragrunt/releases/tag/v0.54.12)
